keyboards/bandominedoni/bandominedoni.h

```diff
@@ -34,12 +34,169 @@
 // Long press: go to _FN layer, tap: MUTE
 #define FN_MUTE  LT(_FN, KC_MUTE)
 #define DF_QWER  DF(_QWERTY)
+#define DF_COLE  DF(_COLEMAK)
 #define MIS_EIS  LT(_MISC,KC_LNG2)
 #define MIS_KAN  LT(_MISC,KC_LNG1)
 
+#define _________________QWERTY_L1_________________ KC_Q,    KC_W,    KC_E,    KC_R,    KC_T
+#define _________________QWERTY_L2_________________ KC_A,    KC_S,    KC_D,    KC_F,    KC_G
+#define _________________QWERTY_L3_________________ KC_Z,    KC_X,    KC_C,    KC_V,    KC_B
+
+#define _________________QWERTY_R1_________________ KC_Y,    KC_U,    KC_I,    KC_O,    KC_P
+#define _________________QWERTY_R2_________________ KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN
+#define _________________QWERTY_R3_________________ KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH
+
+#define _________________COLEMAK_L1________________ KC_Q,    KC_W,    KC_F,    KC_P,    KC_G
+#define _________________COLEMAK_L2________________ KC_A,    KC_R,    KC_S,    KC_T,    KC_D
+#define _________________COLEMAK_L3________________ KC_Z,    KC_X,    KC_C,    KC_V,    KC_B
+
+#define _________________COLEMAK_R1________________ KC_J,    KC_L,    KC_U,    KC_Y,    KC_SCLN
+#define _________________COLEMAK_R2________________ KC_H,    KC_N,    KC_E,    KC_I,    KC_O
+#define _________________COLEMAK_R3________________ KC_K,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH
+
+#define _________________NUMBER_L__________________ KC_1,    KC_2,    KC_3,    KC_4,    KC_5
+// #define _________________NUMBER_R__________________ KC_6,    KC_7,    KC_8,    KC_9,    KC_0
+
+#define _________________FUNC__L___________________ KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5
+// #define _________________FUNC__R___________________ KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10
+
+//  define which MIDI ch to use.
+//  Note that (actual MIDI ch# - 1) -> 0 .. 15 is used for coding.
+//  ch1
+#define DEFAULT_SUB_CH_NUMBER  0
+//  ch2
+#define ALT_SUB_CH_NUMBER      1
+
+enum custom_keycodes {
+    VERSION = QK_KB,
+    TGLCHGR,  //  ToGgLe CH GRoup,
+    MYMOSWP,  //  MY MO SWaP, an alternative of MO_SWAP to enable bellow shake!
+
+   // MY tone to distinguish the notes to avoid sustain effect, etc.
+   // Since the # of codes exceeds the limit of QK_KB range (=64), these are stated here.
+    MY_TONE_MIN  = QK_USER,
+
+    MY_C = MY_TONE_MIN,
+    MY_Cs,
+    MY_Db = MY_Cs,
+    MY_D,
+    MY_Ds,
+    MY_Eb = MY_Ds,
+    MY_E,
+    MY_F,
+    MY_Fs,
+    MY_Gb = MY_Fs,
```

Comment:
Also fairly sure we wont want this being copy pasted to multiple keyboards as it seems to be.

Also use of `QK_USER` at the keyboard level is incorrect. If this is trying to work around some limitation in core, then the fix should be targeted there instead.

Reply:
Thank you for your advice. I will submit a pull request to expand the area for kb.
It used to be kb_user = 256_256, then reallocated to 64_448. (https://github.com/qmk/qmk_firmware/pull/19907)
I'll propose 128_384.

Reply:
Pull request has been issued here (https://github.com/qmk/qmk_firmware/pull/21950).

Reply:
Making the range larger is not the correct change here. #20352 shows you are attempting to duplicate core logic, to fix an issue.  The actual issue with MIDI should be fixed, not this mess of code continually duplicated across multiple keyboards.

Reply:
Oh... I see. You're right.
I'll drop the pull request https://github.com/qmk/qmk_firmware/pull/21950

Then I guess the keyboard matrix location should be considered to differentiate which button was pressed.
I'm not sure if that's possible with my skill...